Measuring a library seems to count its peer dependencies against it

If I clean tmp directory and run 
```
jsize react-dom@next
```

I get an error:

```
✖ 
ERROR in /private/var/folders/bj/0n3sjz9x6qq0s8s8tjfr30r5f42mg6/T/node_modules/react-dom/cjs/react-dom.production.min.js
Module not found: Error: Can't resolve 'react' in '/private/var/folders/bj/0n3sjz9x6qq0s8s8tjfr30r5f42mg6/T/node_modules/react-dom/cjs'
 @ /private/var/folders/bj/0n3sjz9x6qq0s8s8tjfr30r5f42mg6/T/node_modules/react-dom/cjs/react-dom.production.min.js 2:481-497
 @ /private/var/folders/bj/0n3sjz9x6qq0s8s8tjfr30r5f42mg6/T/node_modules/react-dom/index.js
```

As you see it tries to find `react`.

I understand you can’t *bundle* without it, but you probably shouldn’t try to. Since it's a peer dependency, it's unfair to count its size towards the main package. Measuring `react-dom` shouldn't count the size of `react` because the app will only include `react` once.

My intuition is that jsize should read `package.json`, and turn all `peerDependencies` into Webpack `externals` when bundling. Then the size is fair.

--- a/index.js
+++ b/index.js
@@ -65,7 +65,8 @@ function install (id) {
 /**
  * Uses webpack to build a file in memory and return the bundle.
  *
- * @param {*} file - the entry file to build.
+ * @param {*} name - library folder name
+ * @param {*} file - entry point path relative to the library folder
  * @return {Promise<string>}
  */
 function build (name, file) {
@@ -79,7 +80,7 @@ function build (name, file) {
       entry,
       externals,
       plugins: [
-        new webpack.optimize.UglifyJsPlugin({ sourcemap: false }),
+        new webpack.optimize.UglifyJsPlugin({ mangle: false, sourcemap: false }),
         new webpack.DefinePlugin({ 'process.env.NODE_ENV': '"production"', 'process.browser': true })
       ]
     }, (err, stats) => {

--- a/index.js
+++ b/index.js
@@ -21,7 +21,7 @@ function noop () {}
 module.exports = function jsize (pkg) {
   const { name, version, path: file } = parsePackageName(pkg)
   return install(`${name}@${version || 'latest'}`)
-    .then(() => build(path.join(name, file)))
+    .then(() => build(name, file))
     .then(script => {
       const minimized = uglify.minify(script).code
 
@@ -68,14 +68,18 @@ function install (id) {
  * @param {*} file - the entry file to build.
  * @return {Promise<string>}
  */
-function build (file) {
+function build (name, file) {
   return new Promise((resolve, reject) => {
+    const entry = requireRelative.resolve(path.join(name, file), tmp);
+    const packageJson = requireRelative.resolve(path.join(name, 'package.json'), tmp);
+    const externals = Object.keys(require(packageJson).peerDependencies || {});
     const compiler = webpack({
       target: 'web',
       output: { filename: 'file' },
-      entry: requireRelative.resolve(file, tmp),
+      entry,
+      externals,
       plugins: [
-        new webpack.optimize.UglifyJsPlugin({ mangle: true, sourcemap: false }),
+        new webpack.optimize.UglifyJsPlugin({ sourcemap: false }),
         new webpack.DefinePlugin({ 'process.env.NODE_ENV': '"production"', 'process.browser': true })
       ]
     }, (err, stats) => {

--- a/index.js
+++ b/index.js
@@ -80,7 +80,7 @@ function build (name, file) {
       entry,
       externals,
       plugins: [
-        new webpack.optimize.UglifyJsPlugin({ mangle: false, sourcemap: false }),
+        new webpack.optimize.UglifyJsPlugin({ mangle: true, sourcemap: false }),
         new webpack.DefinePlugin({ 'process.env.NODE_ENV': '"production"', 'process.browser': true })
       ]
     }, (err, stats) => {

